x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/circle.test.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act } from 'react-dom/test-utils';
+import { setup, SetupResult, getProcessorValue } from './processor.helpers';
+
+const CIRCLE_TYPE = 'circle';
+
+describe('Processor: Circle', () => {
+  let onUpdate: jest.Mock;
+  let testBed: SetupResult;
+
+  beforeAll(() => {
+    jest.useFakeTimers();
+  });
+
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  beforeEach(async () => {
+    onUpdate = jest.fn();
+
+    await act(async () => {
+      testBed = await setup({
+        value: {
+          processors: [],
+        },
+        onFlyoutOpen: jest.fn(),
+        onUpdate,
+      });
+    });
+    testBed.component.update();
+    const {
+      actions: { addProcessor, addProcessorType },
+    } = testBed;
+    // Open the processor flyout
+    addProcessor();
+
+    // Add type (the other fields are not visible until a type is selected)
+    await addProcessorType(CIRCLE_TYPE);
+  });
+
+  test('prevents form submission if required fields are not provided', async () => {
+    const {
+      actions: { saveNewProcessor },
+      form,
+    } = testBed;
+
+    // Click submit button with only the type defined
+    await saveNewProcessor();
+
+    // Expect form error as "field" is required parameter
+    expect(form.getErrorsMessages()).toEqual([
+      'A field value is required.',
+      'A shape type value is required.',
+    ]);
+  });
+
+  test('saves with required parameter values', async () => {
+    const {
+      actions: { saveNewProcessor },
+      form,
+    } = testBed;
+
+    // Add "field" value (required)
+    form.setInputValue('fieldNameField.input', 'field_1');
+    // Save the field
+    form.setSelectValue('shapeSelectorField', 'shape');
+
+    await saveNewProcessor();
+
+    const processors = getProcessorValue(onUpdate, CIRCLE_TYPE);
+
+    expect(processors[0].circle).toEqual({
+      field: 'field_1',
+      error_distance: 1,
```

Comment:
I noticed that `error_distance` has a default value of 1 in the UI, but it's not mentioned in the docs. I think if `error_distance` doesn't have a default, then its input field should rather stay empty and force the user to set a value. We might need a bug issue for that, but I'd like to confirm with @alisonelizabeth first.

Reply:
Great catch @yuliacech! I didn't notice your review until after mine (I left a similar comment) 😄. 

@danhermann can you help us understand how the `error_distance` option should behave for the `circle` processor? The docs don't mention any default value, but we provide a default value of `1` in the UI. Does this make sense, or should we always force the user to provide their own value?

Reply:
I don't know enough about how the underlying circle shape works to know whether a default value (perhaps scaled by the radius of the circle) for `error_distance` would make sense there or not. Perhaps someone from @elastic/es-analytics-geo make a suggestion?

Reply:
A default value would only make sense if it is scaled to the radius of a circle. Unfortunately, we are setting up processor at this point and have no information about what type of circles the user is going to ingest. So, I think it would be the best to leave it blank.

Reply:
Thanks @imotov for clarifying! I opened https://github.com/elastic/kibana/issues/102147 to address on our side.